source/MaterialXGenMdl/Nodes/ClosureCompoundNodeMdl.h

```diff
@@ -0,0 +1,30 @@
+//
+// TM & (c) 2020 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#ifndef MATERIALX_CLOSURECOMPOUNDNODEMDL_H
+#define MATERIALX_CLOSURECOMPOUNDNODEMDL_H
+
+#include <MaterialXGenMdl/Export.h>
+#include <MaterialXGenMdl/Nodes/CompoundNodeMdl.h>
+
+namespace MaterialX
+{
+
+/// Compound node implementation
+class MX_GENMDL_API ClosureCompoundNodeMdl : public CompoundNodeMdl
+{
+  public:
+    static ShaderNodeImplPtr create();
+
+    void emitFunctionDefinition(const ShaderNode& node, GenContext& context, ShaderStage& stage) const override;
+    void emitFunctionCall(const ShaderNode& node, GenContext& context, ShaderStage& stage) const override;
+
+protected:
+    string _returnStruct;
```

Comment:
It looks like this member variable shadows one with an identical name in `CompundNodeMdl`, so we should either merge them into a single variable, or give this one a unique name.

Reply:
Good catch!